index.js

```diff
@@ -7,12 +7,174 @@ function ask(questionText) {
+    secretNumber = await ask("What is your secret number? I won't peek, I promise...\n");
+    secretNumber = +secretNumber;
+    //error checking
+    while (Number.isInteger(secretNumber) !== true || secretNumber < min || secretNumber > max || secretNumber === 0) {
+      console.log(`Please enter a whole number that's greater than ${min} and less than ${max}.`);
+      secretNumber = await ask(`What would you like your secret number to be?`);
+      secretNumber = +secretNumber;
+    }
+    guess();
+  }
+  if (game === 2) {
+    console.log(`We'll get to this later.`);
+    process.exit();
+  }
 }
+
+async function guess() {
+  let answer = await ask(`Is your number... ${computerGuess}?(Yes or no)\n>_`);
+  count = count + 1;
+  //we wait to sanitize input only once they have evtered a value, because sanitize cannot lowercase undefined.
+  while(answer === ``) {
+    console.log(`Please enter yes or no.`);
+    answer = await ask(`Is your number... ${computerGuess}?\n>_`);
+  }
+  sanitize(answer);
+  //makes sure user input 
+  //while it is not the right guess
+    while (ANSWERS_POS.includes(answer)) {
+      if (secretNumber !== computerGuess) {
+      console.log(`Are you sure about that?`);
+      answer = await ask(`Is your number... ${computerGuess}?\n>_`);
+      sanitize(answer);
+    } if (secretNumber === computerGuess) {
+      computerScore = computerScore + 1;
+      return youWin();
+    }
+  }
+    while (ANSWERS_NEG.includes(answer)) {
+      if (secretNumber !== computerGuess) {
+        return higherLower();
+      }
+      if (secretNumber === computerGuess) {
+        console.log(`Please enter yes or no.`);
+        answer = await ask(`Is your number... ${computerGuess}?\n>_`);
+        sanitize(answer);
+    }
+  }
+}
+
+async function higherLower() {
+  let higher = await ask(`Is it higher (h) or lower (l) than ${computerGuess}?\n>_`);
+  while (higher === ``) {
+    console.log(`Please say if it's higher or lower.`);
+    higher = await ask(`Is it higher or lower than ${computerGuess}?`);
+  }
+  sanitize(higher);
+//we use a while loop to check potential user inputs and then coerce the code into responses based on existing conditions of variables.
+if (secretNumber < computerGuess) {
+  while (ANSWERS_HIGH.includes(higher)) {
+    console.log(`You said it was lower than that.`)
+    higher = await ask(`Is it higher or lower than ${computerGuess}?`);
+    sanitize(higher);
+  } if (ANSWERS_LOW.includes(higher)) {
+    max = computerGuess;
+    computerGuess = Math.floor(split(min, max));
+    return guess();
+}
+}
+if (secretNumber > computerGuess) {
+  while (ANSWERS_LOW.includes(higher)) {
+    console.log(`You said it was higher than that.`)
+    higher = await ask(`Is it higher or lower than ${computerGuess}?`);
+    sanitize(higher);
+  } if (ANSWERS_HIGH.includes(higher)) {
+    min = computerGuess;
+    computerGuess = Math.ceil(split(min, max));
+    return guess();
```

Comment:
These loops and the ones in the guess function are all pretty simple.  You might be able to combine some of the functionality into their own functions, to allow for a little less code that is more readable.  Finding patterns like this is the first step to refactoring, and in that process you usually end up finding a lot of more efficient ways to write the code.